### DISCUSSION TITLE
Write 720KB kryoflux on a 1.44 drive + format needed ?

### DISCUSSION THREAD

**Discussion by retrohospital** [keirf/greaseweazle#436] [Q&A]
Hello everyone !

I'm new to this, and made a few experiments, good and bad, and read the Q&A.

I'm trying to write a 720KB ibm pc floppy image (kryoflux) on a 1.44 HD drive ( a common situation I suppose )

Currently I failed miserably, I have difficulties deciding if I need to choose 80, 84 cylinders/tracks (the image is clealry 80 cyl, 360K per side), and I suppose I need double stepping but I'm not sure of the command line to use.

Also second question ; GW refuses to even start to write my image if I don't format the floppy 720KB, which I find odd for such a low level tool ; why does it need my disk to be formatted ? IS that normal or am I doing something wrong here as well ?

thanks a million,

**Comment by keirf** [comment]
Are you trying to write to a double density disk or is it a high density disk? You don't need to double step -- all 3.5-inch disks are 96 tracks per inch (80 cyl). What image type are you trying to write?

**Comment by retrohospital** [comment]
Hi, I'm trying to write a 720KB kryoflux image on a 720KB disk ( and I'm sure it has never been used on a 1.44 drive - also I can if needed degauss it) on a 1.44 drive

The process goes from start to end but the disk contains random data (directories with funny names, ets) so I guess I'm doing it wrong

Funy thing is, while experimenting I one used to wrong drive (A/B), and sent my 720K kryoflux on my 1.2MB 5 1/4 drive, and this one "successfully" created me a freaking 720KB 5 1/4 floppy, overwriting a 360KB that happened to be in the drive ; which is a bit unexpected. 

I'm trying to do just this but on the good drive, the 3 1/2 one. I tried this on floppies formatted on a real 720KB drive .

ps
Thanks for helping me and really sorry if this is a newbie question, I'm afraid it probably is, I read as much as I could but couldn't really make the existing threads match my use case.

**Reply by keirf** [reply]
Yes there's no real difference between double density 5.25 and 3.5 disks. The tracks are totally interchangeable.

**Reply by keirf** [reply]
If the Kryoflux image is good 720kb image can you try adding `--format=ibm.720` to your command line? This will convert the Kryoflux internally to a normal PC 720kb image and then write it to disk *with verification*. Compare with a raw Kryoflux write which is unverified (write and pray).

**Comment by retrohospital** [comment]
I seem to have difficulties due to the fact my drive is probably exotic.
It's a TEAC FD505, a dual 31/2 et 5/14 drive in a single drive unit.

I often get the "Track0 signal absent after seek to cylinder 0" problem, I often have to power cycle.
I think I need to step slower, it seems to allow me to write my 720K image

**Reply by keirf** [reply]
You could try stepping slower. But the slim mechanisms used in these dual drives often seem to go wrong, especially on the 3.5-inch side.

**Comment by retrohospital** [comment]
Thanks, I'm trying that.

Another issue with the multi drive setup of the combo drive, is that the GW is both a Shugart & IBM device, which is super unusual thing to have on a cable. So if by mistaker you use something else than IBM calls (A/B) it's unpredictable what the drive will do. 
In my case, drives of that erare sitting kind of in the middle as they have jumpers still referring to DS0,DS1 and DS2 but they don't document DS2 at all. 

It is very unclear how an IBM drive behaves if you send shugart signals on the line. (eg select drive DS0 on a twisted cable with an IBM drive)  
Has this been experimented by someone ?
In my case I believe it leaves the drives in an unwanted state and it needs a reboot.

**Reply by keirf** [reply]
The main difference between Shugart and IBM drives is pins 2 and 34. Greaseweazle doesn't usually care about either of those pins.

Usually all IBM drives will be on DS1, then choosing between drives A and B depends on a twist in the cable. Hence Drive B usually is drive select pin 12 and motor pin 16. Drive A is usually after twist and is drive select pin 14 and motor pin 10.

If you want to use IBM addressing, don't jumper a drive other than as DS1!

When only one drive is on the bus, actually `--drive=1` and `--drive=b` are equivalent.

**Comment by retrohospital** [comment]
DO you have a clear idea of what happens when you put an IBM drive on the bus (A/B), and  send instructions to "drive 0" with your GW ?

Again this is a situation that is rather unique to the GW,

**Reply by keirf** [reply]
Pin 10 gets driven as drive select, and pin 16 as motor.

But for an IBM bus, pin 10 is drive A motor and pin 16 is drive B motor. The drive selects are on pins 14 & 12 respectively.

Hence you may find drive motor(s) spin up, but no drive will be selected, and you are sure to get something like a "Track 0 not found" error.

**Comment by retrohospital** [comment]
Interesting, thanks again for your time. 
This is interesting for me because the combo drives offer a lot of jumpers about how they should behave on the line, I can make either drive appear as A or B, strangely enough the jumpers even alow to have both drives appear as the same letter (?!), and they also provide jumper that basically could make them look like shugart devices at ds0,1 and 2 ( but that is not fully documented).

As these devices do not always come with the factory jumper settings after 30+ years, it's important.

This is much more clear to me now, thanks for the support Keir.
